### PR TITLE
Update `mock_call` docs

### LINKED ISSUE #942: Improve `mock_call` docs
Point out:
- `mock_call` mocks only contract entry points
- mocked contract must be deployed by `deploy` or `deploy_contract` cheatcode (related: https://github.com/software-mansion/protostar/issues/944)

### PR DESCRIPTION
**Pull request by Arcticae**
closes #942

### CODE REVIEW

File: website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md [hunk 1]
```diff
@@ -4,7 +4,12 @@
 def mock_call(contract_address: int, fn_name: str, ret_data: Union[List[int], Dict]) -> Callable: ...
 ```
 
-Mocks all calls to function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. Mocked call returns data provided in `ret_data`. Mock works globally, for all of the contracts, not only the testing contract.
+Mocks all calls to a `@view` function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. 
+Mocked call returns data provided in `ret_data`. 
+
+## Caveats
+- The cheatcode works only for contract's exposed `@view`s. Doesn't work for internal functions, `@external`'s etc.
```
Comment (mkaput) [inline website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md:11]:
```suggestion
- This cheatcode works only for contract's exposed `@view`s. It does not work for internal functions, `@external`s etc.
```

**Review by mkaput — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md [hunk 2]
```diff
@@ -4,7 +4,12 @@
 def mock_call(contract_address: int, fn_name: str, ret_data: Union[List[int], Dict]) -> Callable: ...
 ```
 
-Mocks all calls to function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. Mocked call returns data provided in `ret_data`. Mock works globally, for all of the contracts, not only the testing contract.
+Mocks all calls to a `@view` function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. 
+Mocked call returns data provided in `ret_data`. 
+
+## Caveats
+- This cheatcode works only for contract's exposed `@view`s. It does not work for internal functions, `@external`s etc.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md:11]: I dont think it is a correct information. Please check it.

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

**Review by mkaput — APPROVED** [review]

**Review by kasperski95 — APPROVED** [review]

File: website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md [hunk 3]
```diff
@@ -4,7 +4,9 @@
 def mock_call(contract_address: int, fn_name: str, ret_data: Union[List[int], Dict]) -> Callable: ...
 ```
 
-Mocks all calls to function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. Mocked call returns data provided in `ret_data`. Mock works globally, for all of the contracts, not only the testing contract.
+Mocks all calls to a `@view` function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md:7]: This is still incorrect

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md (hunk 3 above)
Comment (Arcticae) [reply website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md:7]: damn, you're right

File: website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md [hunk 4]
```diff
@@ -4,7 +4,11 @@
 def mock_call(contract_address: int, fn_name: str, ret_data: Union[List[int], Dict]) -> Callable: ...
 ```
 
-Mocks all calls to function with the name `fn_name` of a contract with an address `contract_address` until the returned callable is called. Mocked call returns data provided in `ret_data`. Mock works globally, for all of the contracts, not only the testing contract.
+Mocks all calls to function with the name `fn_name` of a contract with an address `contract_address`, until the returned callable is called. 
+
+Mocked call returns data provided in `ret_data`.
+
+Mock works globally, for all contracts, not only the testing contract.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/07-testing/02-cheatcodes/mock-call.md:11]: I mean this PR doesn't change anything rn. Cannot we just close it? (I am ok if we want split it into separate lines)

**Review by kasperski95 — COMMENTED** [review]
Can we highlight somehow this cheatcode doesn't mock internal calls?

**Comment by Arcticae** [comment]
@kasperski95 
[Highlight that mock call works only for the interface functions](https://github.com/software-mansion/protostar/pull/980/commits/d8602c86e9ad1ac870122497df47ccce876f2936)
is that one good enough?

**Review by kasperski95 — APPROVED** [review]

**Pull request merged**